backend/azure/fileSystem.go

```diff
@@ -0,0 +1,138 @@
+package azure
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/c2fo/vfs/v5"
+	"github.com/c2fo/vfs/v5/backend"
+	"github.com/c2fo/vfs/v5/utils"
+)
+
+// Scheme defines the scheme for the azure implementation
+const Scheme = "https"
+
+// Name defines the name for the azure implementation
+const Name = "azure"
+
+const errNilFileSystemReceiver = "azure.FileSystem receiver pointer must be non-nil"
+
+// FileSystem implements the vfs.FileSystem interface for Azure Blob Storage
+type FileSystem struct {
+	options Options
+	client  Client
+}
+
+// NewFileSystem creates a new default FileSystem.  This will set the options options.AccountName and
+// options.AccountKey with the env variables AZURE_STORAGE_ACCOUNT and AZURE_STORAGE_ACCESS_KEY respectively.
+func NewFileSystem() *FileSystem {
```

Comment:
This currently only supports one type of authentication in a fairly rudimentary way.  There is additional work scoped to address the potential of adding additional authentication methods.